### PR TITLE
Implement cross entropy loss

### LINKED ISSUE #588: Implement `CrossEntropyLoss`
I need this for the work on predicting number of halving steps.

### PR DESCRIPTION
**Pull request by joewallwork**
Closes #588 
Merges into #591 

## Checklist

- [x] Implement CrossEntropyLoss
- [ ] ~~Implement all optional arguments~~ Most of them are integer-specific so skipped here
- [x] Add unit tests
- [x] Add CHANGELOG entry
- [x] Update docs

### CODE REVIEW

File: src/ftorch.f90 [hunk 1]
```diff
@@ -55,7 +55,8 @@ module ftorch
                           torch_optim_zero_grad, &  !----------- (procedures)
                           torch_optim_step
 
-  use ftorch_loss, only: torch_loss_mse
+  use ftorch_loss, only: torch_loss_mse, &
+                         torch_loss_cross_entropy
```
Comment (jatkinson1000) [inline src/ftorch.f90:59]: I think I prefer this convention, just an observation that PyTorch uses MSELoss and CrossEntropyLoss.

**Review by jatkinson1000 — APPROVED** [review]
Seems sensible to me pending clarification of expected output size and other docs as discussed in #591 thread.

And assuming rebase/merge prior to merging goes smoothly.
